### PR TITLE
#159965479 Enable Social login.

### PR DESCRIPTION
**Pull request by Teddykavooh**
## What does this PR do?
```
This PR enables both user login and sign up via social login. For example twitter, facebook, github and google.

```

## Description of Task to be completed?
```
If a user want to sign up in Author Haven, all they need to do is click on a link and their details are retrieved fron their respective accounts.

```

## How should this be manually tested?

```
1. Clone this repository
2. Checkout ft-send-email-verification-159965477
3. pip install -r requirements.txt
4. Make migrations and migrate
5. Run the server python manage.py runserver
6. Navigate to 127.0.0.1:8000/api/users/social_auth
7. Enter your provider and access token.
8. Run the post request.
9. Click on your respective account.
10. Data is returned and you are successfully signed up or logged in.

```
## Any background context you want to provide?

```
1. To acces the google access token navigate to https://developers.google.com/oauthplayground/ then click Apps Script API v1 and tick https://www.googleapis.com/auth/userinfo.email. Then click Authorize APIs, click you preffered account and then click on exchange authorization code for token and access your token.
2. To acces the facebook access token navigate to https://developers.facebook.com/apps/296338254283684/dashboard/ then navigate to Roles, Test Users, click on a user's edit, click get an access token, click submit and access your token.
3. To acces the github access token navigate to https://github.com/settings/applications/877190 then click personal access tokens, click generate new token then enter your credentials, give your token description, tick user, click generate token and access your token.
4. To acces the twitter access token and access token secret navigate to https://developer.twitter.com/en/apps/15844618 then click on Keys and tokens.
```

## What are the relevant pivotal tracker stories?

[#159965479](https://www.pivotaltracker.com/n/projects/2192365)

## Screenshots if available to support PR.
screen shot 2018-09-11 at 11 45 12

screen shot 2018-09-11 at 12 45 14

screen shot 2018-09-11 at 11 43 54

screen shot 2018-09-11 at 11 45 19

## Questions:

```
1. What do you think about the social login?
2. Any additional feature you deem important?
```

### CODE REVIEW

**Comment by Quantum-35** [comment]
Resolve the merge conflicts @Teddykavooh

**Review by Quantum-35 — DISMISSED** [review]
Looks good.

**Review by Mutuba — DISMISSED** [review]
Looks good!

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -68,12 +68,21 @@ def validate_password(self, data):
             raise serializers.ValidationError(
                 "Create a password with at least one special character.")
         return data
+    # password = serializers.CharField(
```
Comment (DerKip) [inline authors/apps/authentication/serializers.py]: clean up code that's not in use, #comments

**Review by Quantum-35 — DISMISSED** [review]
LGTM

**Review by Derrickkip — APPROVED** [review]
LGTM

**Review by Quantum-35 — APPROVED** [review]

**Review by WNjihia — APPROVED** [review]

**Pull request merged**